src/Snowflake/Command.php

```diff
@@ -0,0 +1,731 @@
+                                    $tableName
+                                ),
+                                sprintf(
+                                    '%s.%s.%s',
+                                    QueryBuilder::quoteIdentifier($database),
+                                    QueryBuilder::quoteIdentifier($schemaName),
+                                    QueryBuilder::quoteIdentifier($tableName)
+                                ),
+                            ];
+                            return in_array($v['name'], $validSchema);
+                        }
+                    );
+                    $ownershipOnTable = array_filter($tableGrants, fn($v) => $v['privilege'] === 'OWNERSHIP');
+                    assert(count($ownershipOnTable) === 1);
+
+                    $ownershipOnTable = current($ownershipOnTable);
+
+                    self::assignSharePrivilegesToRole($connection, $database, $ownershipOnTable['granted_by']);
+                    self::useRole($connection, $ownershipOnTable['granted_by']);
+
+                    $connection->query(sprintf(
+                        'CREATE TABLE %s.%s.%s AS SELECT * FROM %s.%s.%s;',
+                        QueryBuilder::quoteIdentifier($database),
+                        QueryBuilder::quoteIdentifier($schemaName),
+                        QueryBuilder::quoteIdentifier($tableName),
+                        QueryBuilder::quoteIdentifier($shareDbName),
+                        QueryBuilder::quoteIdentifier($schemaName),
+                        QueryBuilder::quoteIdentifier($tableName),
+                    ));
+
+                    foreach ($tableGrants as $tableGrant) {
+                        self::assignGrantToRole($connection, $tableGrant);
+                    }
+                }
+            }
+        }
+    }
+
+    public static function createUser(
+        LoggerInterface $logger,
+        Connection $connection,
+        array $userGrant,
+        array $passwordOfUsers
+    ): void {
+        self::useRole($connection, $userGrant['granted_by']);
+
+        if (isset($passwordOfUsers[$userGrant['name']])) {
+            $connection->query(sprintf(
+                'CREATE USER %s PASSWORD=\'%s\' DEFAULT_ROLE = %s',
+                $userGrant['name'],
+                $passwordOfUsers[$userGrant['name']],
+                $userGrant['name'],
+            ));
+        } else {
+            $password = Helper::generateRandomString();
+            $logger->alert(sprintf(
+                'User "%s" has been created with password "%s". Please change it immediately!',
+                $userGrant['name'],
+                $password
+            ));
+            $connection->query(sprintf(
+                'CREATE USER %s PASSWORD=\'%s\' DEFAULT_ROLE = %s MUST_CHANGE_PASSWORD = true',
+                $userGrant['name'],
+                $password,
+                $userGrant['name'],
+            ));
+        }
+    }
```

Comment:
Tady ještě bude potřeba nastavit ostatní věci, které ten uživatel má, myslím. 

Má jich spoustu, ale minimálně myslím, že bude potřeba pokud někde má vyplněný DEFAULT WAREHOUSE... Všechny ty property jsou tady: https://docs.snowflake.com/en/sql-reference/sql/create-user.html#syntax

Reply:
![image](https://user-images.githubusercontent.com/642928/215074601-1bf34978-7afe-467f-8baa-4ea2db812727.png)

I když to teda vypadá, že třeba projektový uživatel to vyplněné nemá...

Reply:
hotovo - https://github.com/keboola/project-migration-tool/pull/1/commits/b288c9715a4212b3a9708398a49d3e8ff5a7271a